apps/bestofjs-nextjs/src/app/api/og/projects/route.tsx

```diff
@@ -0,0 +1,52 @@
+import {
+  ProjectPageSearchParams,
+  parseSearchParams,
+} from "@/components/project-list/navigation-state";
+import { api } from "@/server/api-remote-json";
+import { Box, generateImageResponse, mutedColor } from "@/app/api/og/og-utils";
+
+import { ImageLayout } from "../og-image-layout";
+import { ProjectRow } from "../route";
+
+export const runtime = "edge";
+
+export async function GET(_req: Request) {
+  const { tags, query } = useSearchParams(_req.url);
+  const { projects } = await api.projects.findProjects({ query });
```

Comment:
I think you are missing some parameters when calling `api.projects.findProjects()`.
The `query` is about the text search but we have also the "tags", the sort filter, the pagination...

Basically we want to perform the same search as in the regular `/projects` page, using all information provided by the `ProjectSearchQuery` object returned by `parseSearchParams` function.

```ts
  const { tags, sort, page, limit, query } = parseSearchParams(searchParams);
  const sortOption = getSortOption(sort);

  const { projects } = await api.projects.findProjects({
    criteria: tags.length > 0 ? { tags: { $all: tags } } : {},
    query,
    sort: sortOption.sort,
    skip: limit * (page - 1),
    limit,
  });
```

That also means we could do some refactoring to extract the logic shared by the page and the image, since they both need to fetch an array of projects given search parameters, but we can do this later.

Reply:
> I think you are missing some parameters when calling `api.projects.findProjects()`. The `query` is about the text search but we have also the "tags", the sort filter, the pagination...
> 
> Basically we want to perform the same search as in the regular `/projects` page, using all information provided by the `ProjectSearchQuery` object returned by `parseSearchParams` function.
> 
> ```ts
>   const { tags, sort, page, limit, query } = parseSearchParams(searchParams);
>   const sortOption = getSortOption(sort);
> 
>   const { projects } = await api.projects.findProjects({
>     criteria: tags.length > 0 ? { tags: { $all: tags } } : {},
>     query,
>     sort: sortOption.sort,
>     skip: limit * (page - 1),
>     limit,
>   });
> ```
> 
> That also means we could do some refactoring to extract the logic shared by the page and the image, since they both need to fetch an array of projects given search parameters. Basically the snippet above could be inside a shared function), but we can do this later.

I am bit confused as to what is happening here:

```
 const { projects } = await api.projects.findProjects({
    criteria: tags.length > 0 ? { tags: { $all: tags } } : {},
    query,
    sort: sortOption.sort,
    skip: limit * (page - 1),
    limit,
  });
```
Does this object represent the projects filtered by params? 

I agree in extracting the shared logic since a lot of the code I am writing is shared in the page.  Could be good to tackle next!

Reply:
Sorry for the confusing naming, let me describe the parameters of the `findProjects` method:

- `criteria` is an object that expresses the search filter. It uses MongoDB syntax, `$all` being an operator to include elements that all exist inside an array
- `query`: is the text query when users enter some text (I think I should rename it `textQuery` ?)
- `skip` and `limit` is about pagination, I think you are familiar with that if have done SQL requests for example
- `sort` describes the sort to be applied, again using the MongoDB syntax. E.g. `{ stars: -1 }` to sort by stars in the descending order